packages/typespec-azure-resource-manager/src/lib.ts

```diff
@@ -94,6 +94,12 @@ export const $lib = createTypeSpecLibrary({
         default: "Resource types must have a property with '@path` and '@segment' decorators.",
       },
     },
+    "template-type-constraint-no-met": {
+      severity: "error",
+      messages: {
+        default: paramMessage`The template parameter "${"sourceType"}" for "${"entity"}" does not satisfy the constraint type "${"constraintType"}".`,
```

Comment:
```suggestion
        default: paramMessage`The template parameter "${"sourceType"}" for "${"entity"}" does not extend the type "${"constraintType"}". Use the "TrackedResource", "ProxyResource", or "ExtensionResource" template to define the resource.`
```

The important thing is that we give them some actionable thing to do in the message.  If we want this decorator to be generally useful, we might add a parameter to the decorator arguments that includes an optional extra action message, like this added sentence.